ht/services.py

```diff
@@ -38,39 +38,126 @@ def image_result(item):
 				  f"in {search['searchInformation']['formattedSearchTime']}s."
 		)
 		return embed
-	
+
 	pages = tuple(image_result(page) for page in search["items"])
 	await views.Navigator(ctx, pages).run()
 
 async def ds(session, blazon, drawn_kind):
 	blazon_out = urllib.parse.quote(blazon)
 	results = await utils.get_json(session, f"https://drawshield.net/include/drawshield.php?blazon={blazon_out}&outputformat=json")
 	image = discord.File(io.BytesIO(base64.b64decode(results["image"])), filename = "ds.png")
-	
-	embed = embeds.DRAW.create("", f"*{blazon}*", heading = f"{drawn_kind} drawn!")	
-	embed.set_image(url = "attachment://ds.png")	
+
+	embed = embeds.DRAW.create("", f"*{blazon}*", heading = f"{drawn_kind} drawn!")
+	embed.set_image(url = "attachment://ds.png")
 	embed.set_footer(text = f"Not all blazons can be illustrated with this command.\nDrawn using DrawShield; © Karl Wilcox.")
-	
+
 	for message in results["messages"]:
 		if message["category"] != "blazon": continue
 		elif "linerange" in message:
 			embed.add_field(name = f"Error {message['linerange'].strip()}", value = message["content"], inline = False)
 		elif "context" in message:
 			embed.add_field(name = "Error", value = f"{message['content']} {message['context']}", inline = False)
-			
+
 	return embed, image
-	
+
 async def ds_catalog(session, charge):
 	catalog = await utils.get_json(session, f"https://drawshield.net/api/catalog/{urllib.parse.quote(charge)}")
-	
+
 	if not catalog.startswith("http"): return None
 	return catalog.split("\n")
-	
+
 async def commons(session, loop, filename):
 	result_text = await utils.get_text(
 		session,
 		f"https://magnus-toolserver.toolforge.org/commonsapi.php?image={filename}&thumbwidth=600&thumbheight=600&meta"
 	)
 	get_json = lambda text_string, root: ElementTree.fromstring(text_string).find(root)
-	
-	return await loop.run_in_executor(None, get_json, result_text, "file")
\ No newline at end of file
+
+	return await loop.run_in_executor(None, get_json, result_text, "file")
+
+def is_option_keyword(s):
+	return s.startswith(":")
+
+def parse_options_and_blazon(query):
+	words = query.split()
+	options = list(itertools.takewhile(is_option_keyword, words))
+	blazon = " ".join(itertools.dropwhile(is_option_keyword, words))
+	return options, blazon
+
+
+async def heraldicon(session, query):
+	options, blazon = parse_options_and_blazon(query)
+	result = await utils.post_json(session, "https://2f1yb829vl.execute-api.eu-central-1.amazonaws.com/api",
+					{
+						"call": "generate-from-blazon",
+						"data": {
+							"blazon": blazon,
+							"options": options,
+						},
+					})
```

Comment:
```suggestion
	result = await utils.post_json(session, "https://2f1yb829vl.execute-api.eu-central-1.amazonaws.com/api",
		{
			"call": "generate-from-blazon",
			"data": {
				"blazon": blazon,
				"options": options,
			},
		}
	)
```
More in line with my standard code style